tests/order_modifiers_integration/Admin/Modifier_Admin_Handler_Test.php

```diff
@@ -0,0 +1,769 @@
+						'order_modifier_slug'         => 'test_slug_4',
+						'order_modifier_display_name' => 'Test 4 Fee',
+						'order_modifier_form_save'    => true,
+						'modifier'                    => 'fee',
+					],
+					'nonce'          => 'valid_nonce',
+					'valid_nonce'    => true,
+					'expected_error' => 'The field \'status\' is required and cannot be empty.',
+				];
+			},
+		];
+	}
+
+	/**
+	 * @test
+	 * @dataProvider handle_notices_data_provider
+	 */
+	public function it_should_handle_notices_correctly( Closure $fixture ): void {
+		// Capture the success message output.
+		$captured_success_message = null;
+
+		// Mock `render_success_message` to capture the success message.
+		$this->set_class_fn_return(
+			$this->controller_class,
+			'render_success_message',
+			function ( $message ) use ( &$captured_success_message ) {
+				$captured_success_message = $message;
+			},
+			true
+		);
+
+		// Setup the fixture.
+		$data = $fixture();
+
+		// Set request variables for the test.
+		$_GET     = $data['get_data'];
+		$_REQUEST = $data['get_data'];
+
+		// Call the method.
+		$controller = $this->make_controller();
+		$controller->handle_notices();
+
+		// Assert the outcomes.
+		if ( $data['should_display_message'] ) {
+			$this->assertNotNull( $captured_success_message, 'Success message should be displayed.' );
+			$this->assertEquals(
+				__( 'Modifier saved successfully!', 'event-tickets' ),
+				$captured_success_message,
+				'Success message does not match the expected value.'
+			);
+		} else {
+			$this->assertNull( $captured_success_message, 'Success message should not be displayed.' );
+		}
+	}
+
+	public function handle_notices_data_provider(): Generator {
+		yield 'Valid notice display' => [
+			function () {
+				return [
+					'get_data'               => [
+						'updated'  => 1,
+						'edit'     => 1,
+						'modifier' => 'fee',
+						'page'     => 'tec-tickets-order-modifiers',
+					],
+					'should_display_message' => true,
+				];
+			},
+		];
+
+		yield 'Updated flag not set' => [
+			function () {
+				return [
+					'get_data'               => [
+						'edit'     => 1,
+						'modifier' => 'fee',
+						'page'     => 'tec-tickets-order-modifiers',
+					],
+					'should_display_message' => false,
+				];
+			},
+		];
+
+		yield 'Edit flag not set' => [
+			function () {
+				return [
+					'get_data'               => [
+						'updated'  => 1,
+						'modifier' => 'fee',
+						'page'     => 'tec-tickets-order-modifiers',
+					],
+					'should_display_message' => false,
+				];
+			},
+		];
+
+		yield 'Page slug mismatch' => [
+			function () {
+				return [
+					'get_data'               => [
+						'updated'  => 1,
+						'edit'     => 1,
+						'modifier' => 'fee',
+						'page'     => 'invalid_slug',
+					],
+					'should_display_message' => false,
+				];
+			},
+		];
+	}
+
+	/**
+	 * @test
+	 */
+	public function it_should_delete_modifier_correctly(): void {
```

Comment:
For this test, I tried to stay with the mindset of "using what we have available" so instead of manipulating the database, I used `handle_form_submission()` and then parsed the return URl to get the Modifier Id.

What are your thoughts on that?

Reply:
Isn't handle_form_submission manipulating the DB ? Its ok to manipulate the DB as much as we want during a test case. (There are a few exceptions to the as much as we want).

So i would like to see during this test, modifiers being created, ensuring they are, deleted and enduring they are gone.

Reply:
Instead of editing what I had, I decided to create another test that goes through the lifecycle. `it_should_handle_modifier_lifecycle_correctly`